src/bp/chain.rs

```diff
@@ -0,0 +1,1413 @@
+
+    /// Magic number used as prefix in P2P network API
+    pub p2p_magic: P2pNetworkId,
+
+    /// Network name according to BIP 70, which may be different from
+    /// [ChainParams::name]. Not widely used these days, but we still have to
+    /// account for standard.
+    pub bip70_name: String,
+
+    /// HRP bech32 address prefix as defined in BIP 173
+    pub bip173_prefix: String,
+
+    /// Default port for P2P network
+    pub p2p_port: u16,
+
+    /// Default port for full validating node RPC interface
+    pub rpc_port: u16,
+
+    /// Block number from which Lightning network support had started using
+    /// the given chain
+    pub ln_height: u32,
+
+    /// Block number from which RGB had started using the given chain
+    pub rgb_height: u32,
+
+    /// Format of chain-specific data. See [ChainFormat] for more information
+    pub format: ChainFormat,
+
+    /// Dust limit for the given chain; 0 if none dust limit applies
+    pub dust_limit: u64,
+
+    /// Parameters of the native chain asset (can be only one; it is the asset
+    /// in which miners are got paid).
+    pub native_asset: AssetParams,
+
+    /// Flag indicating any kind of testnet network that do not operate with
+    /// real economic values
+    pub is_testnet: bool,
+
+    /// Flag indicating blockchains that use PoW consensus algorithm
+    pub is_pow: bool,
+}
+
+impl PartialEq for ChainParams {
+    fn eq(&self, other: &Self) -> bool {
+        self.genesis_hash == other.genesis_hash
+    }
+}
+
+impl Eq for ChainParams {}
+
+impl StrictEncode for ChainParams {
+    type Error = strict_encoding::Error;
+
+    #[inline]
+    fn strict_encode<E: io::Write>(&self, mut e: E) -> Result<usize, Self::Error> {
+        Ok(strict_encode_list!(e;
+            self.genesis_hash,
+            self.name,
+            self.p2p_magic,
+            self.bip70_name,
+            self.bip173_prefix,
+            self.p2p_port,
+            self.rpc_port,
+            self.ln_height,
+            self.rgb_height,
+            self.format,
+            self.dust_limit,
+            self.native_asset,
+            self.is_testnet,
+            self.is_pow
+        ))
+    }
+}
+
+impl StrictDecode for ChainParams {
+    type Error = strict_encoding::Error;
+
+    #[inline]
+    fn strict_decode<D: io::Read>(mut d: D) -> Result<Self, Self::Error> {
+        Ok(strict_decode_self!(d;
+            genesis_hash,
+            name,
+            p2p_magic,
+            bip70_name,
+            bip173_prefix,
+            p2p_port,
+            rpc_port,
+            ln_height,
+            rgb_height,
+            format,
+            dust_limit,
+            native_asset,
+            is_testnet,
+            is_pow
+        ))
+    }
+}
+
+/// A set of recommended standard networks. Differs from bitcoin::Network in
+/// ability to support non-standard and non-predefined networks
+#[derive(Clone, Debug, Hash)]
+#[non_exhaustive]
+#[repr(u32)]
+pub enum Chains {
```

Comment:
Will it be better to call this `Chain` instead of `Chains`, that might give the false impression that an asset can have multiple chains.

Reply:
You are right, I need to change the name